yang-fetch-imports.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+mod=$1
+
+fatal=0
+if ! which pyang 2>&1 > /dev/null ; then
+  echo "error in $0: pyang required; please install\n(pip install pyang)"
+  fatal=1
+fi
+
+HAS_YANGLINT=1
+if ! which yanglint 2>&1 > /dev/null ; then
+  HAS_YANGLINT=0
+fi
+
+if [ "$fatal" != "0" ]; then
+  exit 1
+fi
+
+set -e
+
+if [ ! -d modules/ ]; then
+  mkdir -p modules/
+fi
+
+before=$(ls modules/ | wc -l)
+after=${before}
+last=$((before-1))
+
+echo "checking for module dependencies (downloading into modules/...)"
```

Comment:
This should write into a dot directory that is added to `.gitignore` by default, but if an environment variable is set, then it should choose that directory instead.  Though the main modules need to be in source control, these extra dependencies should just be downloaded into a scratch space.  Having the environment variable means that you can share a cache between drafts just by setting a value.